hibernate-core/src/main/java/org/hibernate/tool/schema/extract/internal/AbstractInformationExtractorImpl.java

```diff
@@ -999,9 +1002,10 @@ private PrimaryKeyInformation extractPrimaryKeyInformation(
 			final Identifier columnIdentifier = DatabaseIdentifier.toIdentifier(
 					resultSet.getString( getResultSetColumnNameLabel() )
 			);
-			final ColumnInformation column = tableInformation.getColumn( columnIdentifier );
-			pkColumns.add( columnPosition-1, column );
+			fkColumnsByPosition.put( columnPosition - 1, tableInformation.getColumn( columnIdentifier ) );
```

Comment:
Can we please use something like this instead of switching to a `TreeMap`?

```suggestion
			final int index = columnPosition - 1;
			// Fill up the array list with nulls up to the desired index, because some JDBC drivers don't return results ordered by column position
			while ( pkColumns.size() <= index ) {
				pkColumns.add( null );
			}
			pkColumns.set( index, tableInformation.getColumn( columnIdentifier ) );
```

Reply:
done, Thanks